oe_theme.theme

```diff
@@ -690,40 +690,44 @@ function _oe_theme_preprocess_languages(array $link_items, string $current_langu
  * and prepares the links for rendering.
  */
 function oe_theme_preprocess_links__oe_multilingual_content_language_block(array &$variables): void {
-  $entity = \Drupal::service('oe_multilingual.helper')->getEntityFromCurrentRoute();
-  /** @var \Drupal\Core\Entity\EntityInterface $translation */
-  $translation = \Drupal::service('oe_multilingual.helper')->getCurrentLanguageEntityTranslation($entity);
-  $variables['current'] = $translation->language()->getName();
+
+  // Generate the label for the unavailable language.
   /** @var \Drupal\Core\Language\LanguageInterface[] $languages */
   $languages = \Drupal::service('language_manager')->getNativeLanguages();
-  $variables['unavailable'] = $languages[$variables['current_language_id']]->getName();
+  $variables['unavailable_language'] = $languages[$variables['current_language_id']]->getName();
 
   // Normalize the links to an array of optional languages suitable for the ECL.
-  $variables['options'] = [];
+  $variables['languages'] = [];
   foreach ($variables['links'] as $language_code => $link) {
-
-    $language_internal = $language_code;
-    if (EuropeanUnionLanguages::hasLanguage($language_code)) {
-      $language_internal = EuropeanUnionLanguages::getInternalLanguageCode($language_code);
-    }
-
     /** @var \Drupal\Core\Url $url */
     $url = $link['link']['#url'];
     $href = $url
       ->setOptions($link['link']['#options'])
       ->setAbsolute(TRUE)
       ->toString();
 
-    $variables['options'][] = [
+    $variables['languages'][] = [
       'href' => $href,
       'hreflang' => $language_code,
       'label' => $link['link']['#title'],
-      'lang' => $language_internal,
+      'current' => FALSE,
     ];
   }
 
-  // Generate unique element ID for the expandable component.
+  // Add the current language to the list.
+  $entity = \Drupal::service('oe_multilingual.helper')->getEntityFromCurrentRoute();
+  /** @var \Drupal\Core\Entity\EntityInterface $translation */
+  $translation = \Drupal::service('oe_multilingual.helper')->getCurrentLanguageEntityTranslation($entity);
+  $variables['languages'][] = [
```

Comment:
Is it was planned that we use only languages used for content translation only?

Reply:
The functionality is not very clear atm but there is no intention as far as I know to change what we already had, so for now I left it that way and if business wants to add all the languages then we can do it.